### PR TITLE
wideriver: init at 1.1.0

### LINKED ISSUE #307238: Package request: wideriver
**Project description**

Tiling window manager for the [river](https://github.com/riverwm/river) wayland compositor, inspired by [dwm](https://dwm.suckless.org/) and [xmonad](https://xmonad.org/)

**Metadata**

* homepage URL: https://github.com/alex-courtis/wideriver
* source URL: https://github.com/alex-courtis/wideriver
* license: gpl3
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by clebs**
## Description of changes

Closes #307238 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/window-managers/wideriver/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  buildInputs = [ gcc wayland-scanner wayland pkg-config ];
+
+  installPhase = ''
+    export DESTDIR=$out
+    export PREFIX=""
+
+    make install
+  '';
+
+  meta = with lib; {
```
Comment (yunfachi) [inline pkgs/applications/window-managers/wideriver/default.nix:23]:
```suggestion
  meta = {
```
https://github.com/NixOS/nixpkgs/issues/292468

**Review by yunfachi — COMMENTED** [review]
Also `maintainers: add clebs` must be older than `wideriver: init at 1.1.0`.

File: pkgs/applications/window-managers/wideriver/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    description = "Tiling window manager for the river wayland compositor, inspired by dwm and xmonad";
+    homepage  = "https://github.com/alex-courtis/wideriver";
```
Comment (yunfachi) [inline pkgs/applications/window-managers/wideriver/default.nix:25]:
```suggestion
    homepage = "https://github.com/alex-courtis/wideriver";
```
extra space. you can use https://github.com/nix-community/nixpkgs-fmt for convenience

**Comment by clebs** [comment]
@yunfachi do you mean I need to invert the order of my commits? (first `add maintainers` and then `wideriver: init`)

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
Currently, the package is not exposed. Please move it to `pkgs/by-name/wi/wideriver/package.nix`.

(The old way would have been to add the package to `pkgs/top-level/all-packages.nix`, but this pattern is being phased out.)

File: pkgs/by-name/wi/wideriver/package.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  src = fetchFromGitHub {
+    owner = "alex-courtis";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-Jn3DfflDlMSZMlSRBfzQeEUKgYLctd/PiQRWH8uwtlU=";
+  };
+
+  buildInputs = [
+    gcc
```
Comment (tomodachi94) [inline pkgs/by-name/wi/wideriver/package.nix:23]:
```suggestion

```
This doesn't need to be included, it's already included in `stdenv`.

File: pkgs/by-name/wi/wideriver/package.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+  gcc,
```
Comment (tomodachi94) [inline pkgs/by-name/wi/wideriver/package.nix:5]: Remove in tandem with https://github.com/NixOS/nixpkgs/pull/311935#discussion_r1607245223

File: pkgs/by-name/wi/wideriver/package.nix [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  pkg-config,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "wideriver";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "alex-courtis";
+    repo = pname;
```
Comment (tomodachi94) [inline pkgs/by-name/wi/wideriver/package.nix:17]:
```suggestion
    repo = "wideriver";
```
See https://github.com/NixOS/nixpkgs/issues/277994

**Review by tomodachi94 — CHANGES_REQUESTED** [review]

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
Overall looks good, just some additional small things.

**Comment by clebs** [comment]
Thanks for the review and the helpful feedback @tomodachi94!

I moved the package and applied your suggestions.

Additionally, applied `nixmft-rfc-style` and made sure all items in the checklist under https://github.com/NixOS/nixpkgs/tree/master/pkgs#new-packages are met.

File: pkgs/by-name/wi/wideriver/package.nix (hunk 3 above)
Comment (tomodachi94) [reply pkgs/by-name/wi/wideriver/package.nix]: Oops, sorry, I wrote that confusingly. Please undo the `buildInputs` -> `nativeBuildInputs` change.

**Comment by tomodachi94** [comment]
Unable to unresolve a comment, see https://github.com/NixOS/nixpkgs/pull/311935#discussion_r1607278796

File: pkgs/by-name/wi/wideriver/package.nix [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "alex-courtis";
+    repo = "wideriver";
+    rev = version;
+    sha256 = "sha256-Jn3DfflDlMSZMlSRBfzQeEUKgYLctd/PiQRWH8uwtlU=";
+  };
+
+  nativeBuildInputs = [
```
Comment (tomodachi94) [inline pkgs/by-name/wi/wideriver/package.nix:21]:
```suggestion
  buildInputs = [
```
See https://github.com/NixOS/nixpkgs/pull/311935#discussion_r1607278796

**Review by tomodachi94 — CHANGES_REQUESTED** [review]

**Review by tomodachi94 — CHANGES_REQUESTED** [review]
I wrote a comment in a confusing way, sorry!

File: pkgs/by-name/wi/wideriver/package.nix (hunk 6 above)
Comment (clebs) [reply pkgs/by-name/wi/wideriver/package.nix]: Reverted the change!
Will leave resolution open this time 😅

**Review by tomodachi94 — APPROVED** [review]
Lgtm, just make sure to squash down the three `wideriver: ` commits.

**Review by tomodachi94 — APPROVED** [review]
Reapproving :) Thanks for the contribution! ❤️ 🚀

Edit: CI failures do not seem relevant to this PR, so this approval is sustained.

File: pkgs/by-name/wi/wideriver/package.nix [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    owner = "alex-courtis";
+    repo = "wideriver";
+    rev = version;
+    sha256 = "sha256-Jn3DfflDlMSZMlSRBfzQeEUKgYLctd/PiQRWH8uwtlU=";
+  };
+
+  buildInputs = [
+    wayland-scanner
+    wayland
+    pkg-config
```
Comment (Gerg-L) [inline pkgs/by-name/wi/wideriver/package.nix:24]: pkg-config shouldn't be in buildInputs but rather `nativeBuildInputs`

File: pkgs/by-name/wi/wideriver/package.nix [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "wideriver";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "alex-courtis";
+    repo = "wideriver";
+    rev = version;
+    sha256 = "sha256-Jn3DfflDlMSZMlSRBfzQeEUKgYLctd/PiQRWH8uwtlU=";
```
Comment (Gerg-L) [inline pkgs/by-name/wi/wideriver/package.nix:18]:
```suggestion
    hash = "sha256-Jn3DfflDlMSZMlSRBfzQeEUKgYLctd/PiQRWH8uwtlU=";
```

File: pkgs/by-name/wi/wideriver/package.nix [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "wideriver";
+  version = "1.1.0";
+
+  src = fetchFromGitHub {
+    owner = "alex-courtis";
+    repo = "wideriver";
+    rev = version;
```
Comment (Gerg-L) [inline pkgs/by-name/wi/wideriver/package.nix:17]: adding `refs/tags/` removes ambiguity
and use `finalAttrs` to get version
```suggestion
    rev = "refs/tags/${finalAttrs.version}";
```

File: pkgs/by-name/wi/wideriver/package.nix [hunk 10]
```diff
@@ -0,0 +1,45 @@
+{
+  stdenv,
+  fetchFromGitHub,
+  lib,
+  wayland,
+  wayland-scanner,
+  pkg-config,
+}:
+
+stdenv.mkDerivation rec {
```
Comment (Gerg-L) [inline pkgs/by-name/wi/wideriver/package.nix:10]: use the `finalAttrs` pattern
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/wi/wideriver/package.nix [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+
+  meta = {
+    description = "Tiling window manager for the river wayland compositor, inspired by dwm and xmonad";
+    homepage = "https://github.com/alex-courtis/wideriver";
+    license = lib.licenses.gpl3Only;
+    mainProgram = "wideriver";
+    platforms = lib.platforms.linux;
+    maintainers = with lib.maintainers; [ clebs ];
+  };
+}
```
Comment (Gerg-L) [inline pkgs/by-name/wi/wideriver/package.nix:45]:
```suggestion
})
```

File: pkgs/by-name/wi/wideriver/package.nix [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+
+  installPhase = ''
+    runHook preInstall
+
+    export DESTDIR=$out
+    export PREFIX=""
+    make install
+
+    runHook postInstall
+  '';
```
Comment (Gerg-L) [inline pkgs/by-name/wi/wideriver/package.nix:35]:
```suggestion
installFlags = [ "PREFIX=$(out)" ];
```

**Comment by clebs** [comment]
Hi @Gerg-L, 
thanks for the review, I applied all suggestions. PTAL!

**Comment by Gerg-L** [comment]
Result of `nixpkgs-review pr 311935` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wideriver</li>
  </ul>
</details>

**Review by Gerg-L — APPROVED** [review]

**Comment by Gerg-L** [comment]
LGTM!

**Comment by Gerg-L** [comment]
Result of `nixpkgs-review pr 311935` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wideriver</li>
  </ul>
</details>

**Review by Gerg-L — APPROVED** [review]

File: pkgs/by-name/wi/wideriver/package.nix [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-Jn3DfflDlMSZMlSRBfzQeEUKgYLctd/PiQRWH8uwtlU=";
+  };
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    wayland-scanner
```
Comment (Aleksanaa) [inline pkgs/by-name/wi/wideriver/package.nix:26]: Should be in `nativeBuildInputs`?

Comment (clebs) [reply pkgs/by-name/wi/wideriver/package.nix:26]: You are right, since this tool just generates glue code for compilation.

Comment (clebs) [reply pkgs/by-name/wi/wideriver/package.nix]: Moved it, thanks!

**Review by tomodachi94 — APPROVED** [review]
Lgtm per discussion above

**Review by Gerg-L — APPROVED** [review]
LGTM

**Pull request merged**